core/vm/logger.go

```diff
@@ -314,7 +314,8 @@ func (l *StructLogger) CaptureEnter(typ OpCode, from common.Address, to common.A
 		panic("unexpected evm depth in capture enter")
 	}
 	l.statesAffected[to] = struct{}{}
-	theLog := l.logs[lastLogPos]
+	theLog := &l.logs[lastLogPos]
+	theLog.getOrInitExtraData()
```

Comment:
reason to change this?

Reply:
We need to update the log object inside the array (notice the type is []StructLog, not []*StructLog) by `getOrInitExtraData()` method. So it wouldn't take effect if we just copy the object here. "&" indicate the `theLog` variable is just a pointer to the object in the `l.logs` array